### ISSUE TITLE
Auto-Assign Type when a dotAsset is uploaded

### ISSUE THREAD

**Issue by jdotcms** [dotCMS/core#17992]
When a new dotAsset is uploaded into dotCMS, dotCMS will use the values in the mimeType fields to try to intelligently determine the dotAsset type which is being uploaded.

The algo for this decision tree is:

- **Exact/on Site** We have an exact match on the mime or file extension on content types that live on the site that is being uploaded (if we know it) into and to which the user has permissions
sorted alpha

- **Exact/SYSTEM_HOST** Check if we have an exact match on the mime or file extension on content types that live on the SYSTEM_HOST that is being uploaded into and to which the user has permissions

- **Partial Wildcard/on Site** Check if we have a partial wildcard match on the mime on content types that live on the host that is being uploaded into and to which the user has permissions
Partial Wildcard/SYSTEM_HOST Check if we have a partial wildcard match on the mime on content types that live on the SYSTEM_HOST and to which the user has permissions

- **Total Wildcard (or null)/on Site** Check if we have a wildcard match on the mime on content types that live on the host that is being uploaded into and to which the user has permissions

- **Total Wildcard (or null)/SYSTEM_HOST** Check if we have a wildcard match on the mime on content types that live on the SYSTEM_HOST and to which the user has permissions
In all matching cases, if there is only 1 matching Content Type, we will use this one. If there are more than one matches, we will use the first match based on the content type variable name sorted alpha

**Comment by fmontes** [comment]
https://github.com/dotCMS/core/pull/18007

**Comment by bryanboza** [comment]
We have problems trying to use that functionality using the temp multipart resource: 

If you try to send something like this: 
```
curl --location --request PUT 'http://localhost:8080/api/v1/workflow/actions/default/fire/PUBLISH' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic YWRtaW5AZG90Y21zLmNvbTphZG1pbg==' \
--data-raw '{"contentlet":
    {
        "baseType":"dotAsset",
        "hostFolder":"test.com:/images",
        "asset":"temp_b99bb2b932",
        "tags":"test"
    }
}'
```

We are having an error and we are unable to upload the piece of content 
error: `ava.lang.String cannot be cast to java.io.File`

FYI: If I try sending a multipar directly works as expected. 

Here the log with the error : https://gist.github.com/bryanboza/b741ee258f180019137f730cda906614

**Comment by fmontes** [comment]
Move this to an issue: https://github.com/dotCMS/core/issues/18064

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.7 and the main functionality works as expected, the issue will be worked in a separated card  #18064
